### PR TITLE
Consolidate offline and online diagnostics

### PR DESCRIPTION
**Pull request by AnnaKwa**
PR #1343 changed the offline diagnostics metrics calculation to use registries and transforms from the prognostic_run_diags package. These imports are kind of confusing, and since the goal is to try move the offline diags towards using more common code and patterns from the online diags, @nbren12 and I agreed in the discussion over that PR to follow up by having both in a common source tree.

The CLI for the prognostic run report is unchanged.

Refactored public API:
- `offline_ml_diags.compute_diags, offline_ml_diags.create_report` -> `fv3net.diagnostics.offline.compute_diags , fv3net.diagnostics.offline.create_report`

Significant internal changes:
- Reorganization of `prognostic_run_diags` into offline and online diags

The diff is big and hard to go through file by file. Since this is just a reorganization and there are no code changes, the best way to review this is to take a look at the new directory structure for diagnostics here: https://github.com/ai2cm/fv3net/tree/consolidate-diags/workflows/diagnostics

In summary, the relevant directory structure to review is
```
workflows/
- diagnostics/
    - fv3net/
        - diagnostics/
            - _shared/
                - transforms.py
                - registry.py
                - constants.py
            - prognostic_run/
            - offline/
    - tests/
        - offline/
        - prognostic/ 

```

### CODE REVIEW

File: workflows/diagnostics/fv3net/diagnostics/offline/compute_metrics.py [hunk 1]
```diff
@@ -1,8 +1,8 @@
 # TODO: combine offline diags and prognostic diags into same source tree
 # for now, this will import registry and transforms from prognostic diags
-from fv3net.diagnostics.prognostic_run.registry import Registry
-import fv3net.diagnostics.prognostic_run.transform as transform
-from fv3net.diagnostics.prognostic_run.constants import DiagArg
+from fv3net.diagnostics._shared.registry import Registry
```
Comment (oliverwm1) [inline workflows/diagnostics/fv3net/diagnostics/offline/compute_metrics.py:3]: remove TODO above

File: workflows/diagnostics/setup.py [hunk 2]
```diff
@@ -6,6 +6,7 @@
 from setuptools import setup, find_namespace_packages
 
 install_requirements = [
+    "scipy>=1.5.0",
```
Comment (oliverwm1) [inline workflows/diagnostics/setup.py:9]: fv3fit and loaders are also now requirements for this package.

File: docs/packages.rst [hunk 3]
```diff
@@ -42,7 +42,7 @@ fv3config_ (`docs <https://fv3config.readthedocs.io/en/latest/>`_) provides rout
     readme_links/fv3fit_readme
     readme_links/loaders_readme
     readme_links/fv3viz_readme
-    readme_links/offline_ml_diags_readme
+    readme_links/offline_readme
```
Comment (oliverwm1) [inline docs/packages.rst:45]: does this offline readme still exist somewhere?

File: workflows/diagnostics/README.md [hunk 4]
```diff
@@ -4,7 +4,7 @@ Compute summary diagnostics and metrics from the output of a prognostic run and
 visualize on report comparing multiple runs.
 
 See the [Sphinx
-Documentation](https://vulcanclimatemodeling.com/docs/prognostic_run_diags/)
+Documentation](https://vulcanclimatemodeling.com/docs/diagnostics/)
```
Comment (oliverwm1) [inline workflows/diagnostics/README.md:7]: update this README to reflect that this package now does offline diagnostics too. Maybe could just copy in the offline README to the end of this one, and leave the Sphinx docs to just describe the online diags?

File: workflows/diagnostics/fv3net/diagnostics/offline/compute_metrics.py [hunk 5]
```diff
@@ -1,8 +1,8 @@
...
+import fv3net.diagnostics._shared.transform as transform
+from fv3net.diagnostics._shared.constants import DiagArg
```
Comment (oliverwm1) [inline workflows/diagnostics/fv3net/diagnostics/offline/compute_metrics.py:5]: I think the `constants` module should be split into constants that are actually shared between the workflows (just `DiagArg`?) and leave a module within `prognostic_run` for the constants that are specific to that workflow.

**Review by oliverwm1 — COMMENTED** [review]
Generally looks good! This helps make the fv3net repo easier to explore I think. I do have some minor comments.

**Comment by nbren12** [comment]
Thanks. just one name quibble: `fv3net.diagnostics.offline.compute_diags` is a bit redundant suggest renaming to `fv3net.diagnostics.offline.compute`.

File: docs/packages.rst (hunk 3 above)
Comment (AnnaKwa) [reply docs/packages.rst:45]: Oops, it got lost in the move... I put it back in the offline dir.

**Comment by AnnaKwa** [comment]
Thanks @oliverwm1 for reviewing, I addressed the comments and this is ready for re-review now.

**Comment by oliverwm1** [comment]
> Thanks @oliverwm1 for reviewing, I addressed the comments and this is ready for re-review now.

Looks to me like the `_shared.constants` module still contains a bunch of prognostic run specific constants:

https://github.com/ai2cm/fv3net/blob/consolidate-diags/workflows/diagnostics/fv3net/diagnostics/_shared/constants.py

**Review by oliverwm1 — APPROVED** [review]
LGTM!
